### ISSUE TITLE
relative/absolute URL check seems broken

### ISSUE THREAD

**Issue by mozfreddyb** [mozilla/addons-linter#2954]
### Describe the problem and steps to reproduce it:

It seems to me that [isRelativeUrl()](https://github.com/mozilla/addons-linter/blob/c47afc30d6462797d365d438557ae38e41acc287/src/schema/formats.js#L64) never returns true. It's worse than a security check that never returns false, but it's still pretty wrong.

### What happened?

I tried with `isRelativeUrl("foo.js")` and get `undefined`.

### What did you expect to happen?

In this case, I'd like it to return `true` for relative URLs. 

### Anything else we should know?

More generally speaking, I would like to suggest to use a whatwg-spec compliant URL parsing API, as they also allow [resolving URLs given a _base_ domain](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#Parameters). That would allow resolving all specified URLs given a (mock) moz-extension API and then checking whether they are still pointing within that extension.

something along the lines of 
```
// suppliedURL is the input here
let baseURLString = 'moz-extension://abcdefgh-1337-1234-4321-abcdef012345/';
let baseURL = new URL(baseURL);
let fullURL = new URL(suppliedURL, baseURLString);
// now you can check wether `fullURL` has the same `origin` attribute as `baseURL` to see if a URL is remote.
}
```

Functions like `isSecure` could check the `protocol` attribute.
The `data` URL checks would have to check the protocol and then do some hackery, unfortunately..

**Comment by marchiore** [comment]
Hey @mozfreddyb, can i work on this?

**Comment by marchiore** [comment]
I took the liberty to look into this problem but i can't reproduce the`undefined` return on `isRelativeUrl("foo.js")`.

Steps that i made trying to reproduce the problem:

### Created a simple test case

```js
it('relative url check', () => {
    const value = 'foo.js';

    expect(isRelativeUrl(value)).toEqual(true);
});
```
### Run it

I used the existing test `test.formats.js` to include this one and after run with `jest test.formats.js`

it pass without problems

<img width="502" alt="image" src="https://user-images.githubusercontent.com/648079/79146213-63bc3080-7d98-11ea-802c-022657657bab.png">

I'm missing something here? How can i reproduce the error?

**Comment by mozfreddyb** [comment]
@marchiore This is not my repository, I am merely the reporter. 

I think we need @EnTeQuAk or @rpl to respond.

**Comment by Rob--W** [comment]
The implementation of `isRelativeUrl` cannot return `undefined`, as all exiting branches are guaranteed to return booleans. The current implementation is still the same as the time of the report, so I guess that a mistake was made when the bug was filed.

https://github.com/mozilla/addons-linter/blob/c47afc30d6462797d365d438557ae38e41acc287/src/schema/formats.js#L52-L79

@mozfreddyb Could you double-check whether your issue description matches with the bug that you intended to report?

**Comment by mozfreddyb** [comment]
Indeed, I can not verify this anymore. Weird! :disappointed:

**Comment by Rob--W** [comment]
Thanks for your confirmation. I'll close the issue then.
